phoenix-core/src/it/java/org/apache/phoenix/end2end/UpsertWithSCNIT.java

```diff
@@ -0,0 +1,167 @@
+package org.apache.phoenix.end2end;
+
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+public class UpsertWithSCNIT extends ParallelStatsDisabledIT {
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+
+
+    @Test // See https://issues.apache.org/jira/browse/PHOENIX-4983
+    public void testUpsertOnSCNSetTxnTable() throws SQLException {
+
+        String trnx = "CREATE TABLE transaction_table (METRIC_ID CHAR(15) NOT NULL,\n" +
```

Comment:
Use a generateUniqueName() to create tables with unique names since tests that extend ParallelStatsDisabledIT 
 share the same cluster.